packages/oauth/src/core/oauth2.ts

```diff
@@ -155,6 +155,55 @@ export const generateState = () => {
 	return generateRandomString(43);
 };
 
+export const refreshOAuth2Tokens = async <_ResponseBody extends {}>(
```

Comment:
Refreshing access tokens isn't part of the OAuth spec, so I don't think we should add it to the core

Reply:
It would be useful in the future if lucia want to add other self hosted Authentication provider as Authentical or Authelia.

Reply:
The alternative is for Lucia to not handle it at all, since it’s out of scope of the package (see other review)

Reply:
I am currently replacing next-auth with lucia and I had the token refresh flow with keycloak working in next-auth. I think the best solution would be to not include it in lucia but to write a small instruction in the docs on how to set it up. Having a lucia oauth keycloak only setup in next.js where would I place the refresh code?

Reply:
Lucia oauth is deprecated. Oauth will move to "arctic" library, also with refresh token method.
Hopefully it will support keycloak as next provider.

Reply:
Well yeah, I just read that and I have to say that I love everything that comes with v3. There unfortunately it is not an official release yet and I am in the process of implementing all the features I had working with next-auth. Lucia is just a better fit for me. 
If there is a experimental version with keycloak I could try out I would be interested to do so otherwise I would just glue some access token refresh code to the current version and migrate as soon as there is something.